mobile/src/components/common/button/button.tsx

```diff
@@ -1,26 +1,58 @@
 import React, { FC } from 'react';
 
-import { ButtonVariant } from '~/common/enums/enums';
+import { AppColor, ButtonVariant } from '~/common/enums/enums';
 import { IconName } from '~/common/types/types';
-import { Icon, Pressable, Text } from '~/components/common/common';
+import {
+  Icon,
+  Pressable,
+  Spinner,
+  Text,
+  View,
+} from '~/components/common/common';
 
 import { styles } from './styles';
 
 type Props = {
   label: string;
   icon?: IconName;
   variant?: ButtonVariant;
+  size: 'small' | 'large';
   onPress: () => void;
+  isLoading?: boolean;
 };
 
-const Button: FC<Props> = ({ label, icon, variant, onPress }) => {
+const Button: FC<Props> = ({
+  label,
+  icon,
+  variant,
+  onPress,
+  size,
+  isLoading,
+}) => {
   const textMarginLeft = icon ? 10 : 0;
+  const loaderColor =
+    variant === ButtonVariant.PRIMARY
+      ? AppColor.TEXT.GRAY_100
+      : AppColor.BRAND.BLUE_100;
+
+  const sizeStyle = size === 'small' ? 'buttonSmall' : 'buttonLarge';
```

Comment:
I think it's better to do something like this: 
```suggestion
  const sizeStyle = `button${capitalize(size)}`
```